backend/company/models.py

```diff
@@ -11,9 +11,14 @@ class Address(models.Model):
     def __str__(self):
         return f"{self.street}, {self.zip_code} {self.city}"
 
+    def update(self, validated_data):
+        for attr, value in validated_data.items():
+            setattr(self, attr, value)
+        self.save()
+
 
 class Company(models.Model):
-    ico = models.IntegerField(blank=False, null=False, primary_key=True)
+    ico = models.CharField(blank=False, null=False, primary_key=True, max_length=8)
```

Comment:
I changed field to CharField, because ICO can start with zero, which is not valid integer